### DISCUSSION TITLE
Dynamic configuration support at runtime

### DISCUSSION THREAD

**Discussion by dariusclay** [open-telemetry/opentelemetry-specification#3445] [Q&A]
Hi,
Are there any plans to allow dynamic configuration of Otel instrumentation? For example being able to change which trace sampler is used, or configure the probability for a TraceIdRatioBased sampler. As I understand it now, once it's configured you cannot change it without a restart of the service.

**Comment by tsloughter** [comment]
There is no active work in this area at the moment. For just sampling you should check out the JaegerRemoteSampler, https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#jaegerremotesampler

**Reply by tsloughter** [reply]
Oh, I'll also point to the current work on file configuration where future dynamic configuration is mentioned https://github.com/open-telemetry/opentelemetry-specification/pull/3437/files#diff-de4a069dd0a4c04715073ed9eba492f6cb9e4c4c3101761fddd3c840941a2064R109
